presto-main/src/main/java/com/facebook/presto/sql/planner/plan/AggregationNode.java

```diff
@@ -71,6 +71,26 @@ public boolean isOutputPartial()
         {
             return outputPartial;
         }
+
+        public static Step partialOutput(Step step)
```

Comment:
It looks like this method is only used in the operator, so I would move the logic there